### PR TITLE
HSIEO-5499: Fix build error (unnecessary usage of removed class IProgressListener)

### PR DESCRIPTION
**Pull request by isc-bzhang**

### CODE REVIEW

File: src/cls/_ZPM/DependencyAnalyzer.cls
Comment (isc-kiyer) [inline src/cls/_ZPM/DependencyAnalyzer.cls:35]: Shouldn't be removed. Should be replaced with %ZHSLIB.PackageManager.Developer.Extension.PackageManager.UI.DependencyAnalyzerSocket.

If that class will be in the separate zpm-studio-ui package, may want to actually add IProgressListener back as %ZPM.General.IProgressListener perhaps.

**Review by isc-kiyer — CHANGES_REQUESTED** [review]
Just the one note. Need to make a decision as to how to handle it

File: src/cls/_ZPM/DependencyAnalyzer.cls
Comment (isc-bzhang) [reply src/cls/_ZPM/DependencyAnalyzer.cls:35]: @isc-kiyer Yeah so the UI.DependencyAnalyzerSocket is removed and will not be presented in zpm-studio-ui package as well, because we are not supporting the dependency analyzer studio UI for IPM. In that case, is fine to just delete the IProgressListener here (as the original class is just an abstract cls)? Or actually should wejust remove the ZPM.DependencyAnalyzer class as well since we're not using the dependencyAnalyzer UI?

Comment (isc-kiyer) [reply src/cls/_ZPM/DependencyAnalyzer.cls:35]: Ah ok in that case, just update references to the method call. If there are currently none in zpm, copy the dependency analyzer command line command we added recently in ZHSLIB so that the dependency analyzer is still available in some form.

Comment (isc-kiyer) [reply src/cls/_ZPM/DependencyAnalyzer.cls:35]: Also, in that case, can just write out the statements rather than logging them to the progress listener for printing all at once later/in a UI.

Comment (isc-bzhang) [reply src/cls/_ZPM/DependencyAnalyzer.cls:35]: @isc-kiyer Sounds good, I will put new dependencyAnalyzer calls to %ZPM.Main.cls. Additionally, for calls in this class, in addition to replacing IProgressListener usage with Write statements, I think we still needs to modify those method arguments, since IProgressListener class is now removed, if we still keep its type, the class would report error during build compilation

Comment (isc-kiyer) [reply src/cls/_ZPM/DependencyAnalyzer.cls:35]: Great! Yes I agree need to remove all references to IProgressListener

Comment (isc-bzhang) [reply src/cls/_ZPM/DependencyAnalyzer.cls:35]: @isc-kiyer  Add & remove done; new jira here: https://usjira.iscinternal.com/browse/HSIEO-5548

**Comment by isc-bzhang** [comment]
Merging this without approval to unblock the work for IPM team

**Comment by isc-kiyer** [comment]
@Beiming-Zhang Will you handle using the verbose flag + optionally giving a file path to write to in another PR? If so, please create a Jira and note it here.

**Pull request merged**
